lib/MwbExporter/Formatter/Doctrine2/Annotation/Model/Column.php

```diff
@@ -158,10 +152,21 @@ public function writeGetterAndSetter(WriterInterface $writer)
      */
     public function asAnnotation()
     {
+        // Preserved for backwards-compatibility, but we can also use the Doctrine DC2Type annotations
+        // to indicate doctrine types in field comments (e.g. "(DC2Type:guid)", "(DC2Type:json)", etc).
+        if ($this->isUuid()) {
+            return ['type' => 'guid', 'unique' => true];
+        }
+
         $attributes = array(
             'name' => ($columnName = $this->getTable()->quoteIdentifier($this->getColumnName())) !== $this->getColumnName() ? $columnName : null,
             'type' => $this->getFormatter()->getDatatypeConverter()->getMappedType($this),
         );
+
+        if (($type = $this->extractDoctrineType())) {
```

Comment:
This is neat but a little misleading for the unwary - the statement is not comparing the value of $type with return from function rather it is assigning $type and checking truthiness  all in one.
I would prefer an explicit $type = $this->extractDoctrineType();
followed by if( !is_null($type) ){ ...

Reply:
objective c brackets emphasise assignment